velox/connectors/tpch/TpchConnector.h

```diff
@@ -140,12 +140,10 @@ class TpchConnector final : public Connector {
       const std::unordered_map<
           std::string,
           std::shared_ptr<connector::ColumnHandle>>& columnHandles,
-      ConnectorQueryCtx* FOLLY_NONNULL connectorQueryCtx) override final {
+      ConnectorQueryCtx* FOLLY_NONNULL connectorQueryCtx,
+      const dwio::common::ReaderOptions& options) override final {
     return std::make_unique<TpchDataSource>(
-        outputType,
-        tableHandle,
-        columnHandles,
-        connectorQueryCtx->memoryPool());
+        outputType, tableHandle, columnHandles, options);
```

Comment:
Don't we have to set options->setMemoryPool(connectorQueryCtx->memoryPool()) before returning here as well?

Reply:
I am not sure but I would expect the creator of the ReaderOptions to have done that already. In fact TableScan.cpp line 111 does just that.